src/primitives/sphere.jl

```diff
@@ -37,6 +37,30 @@ end
 
 Sphere(p1::Tuple, p2::Tuple, p3::Tuple) = Sphere(Point(p1), Point(p2), Point(p3))
 
+"""
+    Sphere(p1, p2, p3, p4)
+
+A 3D sphere passing through points `p1`, `p2`, `p3` and `p4`.
+"""
+function Sphere(p1::Point{3}, p2::Point{3}, p3::Point{3}, p4::Point{3})
+  v1 = p1 - p4
+  v2 = p2 - p4
+  v3 = p3 - p4
+  d = det(hcat(v1, v2, v3))
+  T = typeof(d)
+  if isapprox(d, zero(T); atol = atol(T))
+    error("The four points are coplanar.")
```

Comment:
```suggestion
    throw(ArgumentError("four points are coplanar"))
```
Please follow previous review suggestions. We try to throw specific errors usually.